### ISSUE TITLE
Core dump in test_c_api.test_generic_filter

### ISSUE THREAD

**Issue by matthew-brett** [scipy/scipy#7580]
There appears to be a new core dump in the scipy wheel daily tests, for Python 3.6 on Linux (EDIT and OSX):

```
test_c_api.test_generic_filter(0,) ... *** Error in `python': free(): invalid size: 0x00007f5d2cee61e0 ***
config.sh: line 56:    50 Aborted                 (core dumped) python -c "$test_cmd"
```

https://travis-ci.org/MacPython/scipy-wheels/jobs/251928393
https://travis-ci.org/MacPython/scipy-wheels/jobs/251928396

@jaimefrio - could this be the ndimage refactor?

**Comment by matthew-brett** [comment]
Also on OSX: 

https://travis-ci.org/MacPython/scipy-wheels/jobs/251928415

**Comment by jaimefrio** [comment]
That speaks loudly of #7570, let me take a look.

Can you share specifics about those build settings? I test it locally on OSX, and Travis didn't complain, so it is kind of surprising that this keeps happening.

**Comment by matthew-brett** [comment]
Yes - the build settings are in the config.sh file:

https://github.com/MacPython/scipy-wheels/blob/master/config.sh

It's a little bit more complicated than it could be, for OSX, because it has echoes of the previous dual arch build that we abandoned a few months ago because no-one cared about 32-bit.  Then again, it's really bare-bones for manylinux, which is also failing.

Are you testing Python 3.6 on your local machine?  I guess so, but just to be sure.

**Comment by jaimefrio** [comment]
Is 3.6 noticeably different from 3.5? Because that's what I have been testing locally...

**Comment by matthew-brett** [comment]
The core dumps are specific to Python 3.6, I don't know why - here's the full grid:

https://travis-ci.org/MacPython/scipy-wheels/builds/251928376

(there's a different, intermittent Python 2.7 failure).   I can't explain why 3.6 doesn't fail on the standard scipy test grid, unless it's some interaction with OpenBLAS (I think the standard tests use ATLAS).

**Comment by matthew-brett** [comment]
By the way - thanks very much for taking this on, it's great to see ndimage get some love.

**Comment by jaimefrio** [comment]
I'm not sure I'm going to be much of a loving partner by the time this is over...;-)

**Comment by jaimefrio** [comment]
It seems to be the interaction of Python 3.6 with NumPy 1.11.3 that's causing this. It doesn't happen with latest numpy...

**Comment by jaimefrio** [comment]
Should be fixed in #7581, do you have an easy way to verify it, @matthew-brett?

**Comment by matthew-brett** [comment]
I do - running here: https://travis-ci.org/MacPython/scipy-wheels/builds/252210561
